### DISCUSSION TITLE
org-modern breaks on blocks that are rendered later

### DISCUSSION THREAD

**Discussion by acid-bong** [minad/org-modern#218] [Q&A]
I'm using save-place-mode, so I often spawn at the end or in the middle of my `config.org`. This way I scroll bottom-to-top, and blocks aren't fontified until i reach `#+begin_src`. When it only appears in the view, the end marker is replaced correctly.
![Screenshot_2024-06-30-09-42-02-660_com termux-edit](https://github.com/minad/org-modern/assets/94849097/16483567-8420-49fe-a44d-8a8d7a2ef10c)

But as soon as I scroll to the top of the block, the replacement disappears, and the markers no longer look symmetrical.
![Screenshot_2024-06-30-09-42-31-948_com termux-edit](https://github.com/minad/org-modern/assets/94849097/3a613ff6-c2f8-4a42-9e12-f6dfabd88c68)

These screenshots are from TUI Emacs in Termux, the theme is ef-dark by Prot. The same happens on my desktop, with GUI Emacs and doom-gruvbox

**Comment by minad** [comment]
Hi, I've also seen this occasionally. The problem is not an Org-modern bug, since it also occurs if you disable `global-org-modern-mode`. If you place point within a large source block region, close the file and reopen it again (with `save-place-mode`), the block will not be fontified correctly. The problem seems to be that during fontification of large regions, an extension of the highlighted region has to be performed first (`org-fontify-extend-region`) and this code has a bug.

**Reply by acid-bong** [reply]
So it's an Emacs issue after all

**Reply by minad** [reply]
Yes, can you please report this problem to the Org developers?
